smsdk/client.py

```diff
@@ -225,3 +229,118 @@ def get_machine_schema(self, machine_source, types=[], return_mtype=False, **kwa
+        then see also get_machine_names(). 
+
+        :param normalize: Flatten nested data structures
+        :type normalize: bool
+        :return: pandas dataframe
+        """
+        return self.get_data_v1('machine_v1', 'get_machines', normalize, *args, **kwargs)
+
+    def _get_machine_types(self, normalize=True, *args, **kwargs):
+        """
+        Get list of machine types and associated metadata.  Note this includes extensive internal metadata.  If you only want to get a list of machine type names
+        then see also get_machine_type_names(). 
+
+        :param normalize: Flatten nested data structures
+        :type normalize: bool
+        :return: pandas dataframe
+        """
+
+        return self.get_data_v1('machine_type_v1', 'get_machine_types', normalize, *args, **kwargs)
+    
+    def get_factories(self, normalize=True, *args, **kwargs):
+        generator = self._get_factories(normalize=normalize, *args, **kwargs)
+        data = []
+        for page in generator:
+            try:
+                data.append(page)
+            except Exception as e:
+                print(e)
+        data = pd.concat(data)
+        return data
+
+    def get_machines(self, normalize=True, *args, **kwargs):
+        generator = self._get_machines(normalize=normalize, *args, **kwargs)
+        data = []
+        for page in generator:
+            try:
+                data.append(page)
+            except Exception as e:
+                print(e)
+        data = pd.concat(data)
+        return data
+
+    def get_machine_types(self, normalize=True, *args, **kwargs):
+        generator = self._get_machine_types(normalize=normalize, *args, **kwargs)
+        data = []
+        for page in generator:
+            try:
+                data.append(page)
+            except Exception as e:
+                print(e)
+        data = pd.concat(data)
+        return data
+
+    def get_machine_names(self, source_type=None, clean_strings_out=True):
+        """
+        Get a list of machine names.  This is a simplified version of get_machines().  
+
+        :param source_type: filter the list to only the specified source_type
+        :type source_type: str
+        :param clean_strings_out: If true, return the list using the UI-based display names.  If false, the list contains the Sight Machine internal machine names.
+        :return: list
+        """
+
+        query_params = {'_only': ['source', 'source_clean', 'source_type'],
+                        '_order_by': 'source_clean'}
+
+        if source_type:
+            # Double check the type
+            mt = self.get_machine_types(source_type=source_type)
+            # If it was found, then no action to take, otherwise try looking up from clean string
+            mt = self.get_machine_types(source_type_clean=source_type) if not len(mt) else []
```

Comment:
duplicate calls for the same functionality

Reply:
for mt it is required, we are comparing against source_type and source_type_clean